ister.py

```diff
@@ -87,30 +87,22 @@ def run_command(cmd, raise_exception=True, log_output=True, environ=None,
                                 stderr=subprocess.PIPE,
                                 env=environ,
                                 shell=shell)
-        lines = proc.stdout
-        stdout = []
-        for line in lines:
-            decoded_line = line.decode('ascii', 'ignore').rstrip()
-            stdout.append(decoded_line)
-            if show_output:
-                LOG.info(decoded_line)
-            elif log_output:
-                LOG.debug(decoded_line)
-        lines = proc.stderr
-        stderr = []
-        for line in lines:
-            decoded_line = line.decode('ascii', 'ignore').rstrip()
-            stderr.append(decoded_line)
-            if show_output:
-                LOG.info(decoded_line)
-            elif log_output:
-                LOG.debug(decoded_line)
+        outputs = []
+        for lines in (proc.stdout, proc.stderr):
+            output = []
+            for line in lines:
+                decoded_line = line.decode('ascii', 'ignore').rstrip()
+                output.append(decoded_line)
+                if show_output:
+                    LOG.info(decoded_line)
+                elif log_output:
+                    LOG.debug(decoded_line)
```

Comment:
I think you wanted outputs.append(output) after the inner for loop.

Reply:
Sorry, I fixed this all yesterday and tested this patch as well. But I did not push out and you ended up looking at the initial version.